### PR TITLE
Add IERC20Metadata with name, symbol and decimals

### LINKED ISSUE #2559: Add getter for decimals to IERC20.sol
**🧐 Motivation**
Could use a getter for decimals()
**📝 Details**
IERC20 = IERC20(address);
uint decimals = token.decimals();

### PR DESCRIPTION
**Pull request by Amxx**
Fixes #2559

#### PR Checklist
- [x] Documentation
- [x] Changelog entry

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
  * Rename `UpgradeableProxy` to `ERC1967Proxy`. ([#2547](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2547))
  * `ERC777`: Optimize the gas costs of the constructor. ([#2551](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2551))
  * `ERC721URIStorage`: Add a new extension that implements the `_setTokenURI` behavior as it was available in 3.4.0. ([#2555](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2555))
+ * `ERC20Metadata`: Add a new extended interface that includes the non standard `name()`, `symbol()` and `decimals()` functions. ([#2561](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2561))
```
Comment (frangio) [inline CHANGELOG.md:21]:
```suggestion
 * `IERC20Metadata`: Add a new extended interface that includes the optional `name()`, `symbol()` and `decimals()` functions. ([#2561](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2561))
```

File: contracts/token/ERC20/extensions/IERC20Metadata.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "../IERC20.sol";
+
+/**
+ * @dev Interface of the ERC20 standard as defined in the EIP.
```
Comment (frangio) [inline contracts/token/ERC20/extensions/IERC20Metadata.sol:8]:
```suggestion
 * @dev Interface for the optional metadata functions from the ERC20 standard.
```

File: contracts/token/ERC20/extensions/IERC20Metadata.sol [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ */
+interface IERC20Metadata is IERC20 {
+    /**
+     * @dev Returns the name of the tokens.
```
Comment (frangio) [inline contracts/token/ERC20/extensions/IERC20Metadata.sol:12]: I think "tokens" should be singular "token" here and in the other functions.

**Review by frangio — COMMENTED** [review]
Thanks! We need to add this in `tokens/ERC20/README.adoc` as well.

**Comment by Amxx** [comment]
@frangio Should this be called IERC20Extended ?

**Comment by frangio** [comment]
I thought `IERC20Metadata` was good! Alternatively I'd go with `IERC20Optional`.

**Comment by Amxx** [comment]
> I thought `IERC20Metadata` was good! Alternatively I'd go with `IERC20Optional`.

ok, that is reverted

**Review by frangio — APPROVED** [review]
Since this is not going into 4.0, I moved the changelog entry to a new section.

**Pull request merged**